checks.d/process.py

```diff
@@ -249,14 +251,18 @@ def check(self, instance):
             raise KeyError('The "name" of process groups is mandatory')
 
         if search_string is None:
-            raise KeyError('The "search_string" is mandatory')
-
-        pids = self.find_pids(
-            name,
-            search_string,
-            exact_match,
-            ignore_ad=ignore_ad
-        )
+            if pid is not None:
+                pids = [psutil.Process(pid)]
+            else:
+                raise KeyError('The "search_string" is mandatory')
```

Comment:
That shouldn't be a KeyError. That could be a ValueError or just a bare Exception